### PR TITLE
Use MKL Rewrite pass in convolution microbenchmark

### PR DESCRIPTION
**Pull request by milpuz01**
Instead of hard-coding whether to use Eigen or MKL node in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/conv_ops_benchmark_test.cc leverage MklLayoutRewritePass class to decide whether to rewrite node or not. 

This also aligns with change present in this PR: https://github.com/tensorflow/tensorflow/pull/60026 where decision whether to rewrite node is decided based if computation cost amortizes overhead cost.

### CODE REVIEW

File: tensorflow/core/kernels/conv_ops_benchmark_test.cc [hunk 1]
```diff
@@ -97,19 +95,45 @@ static Conv2DGraph Conv2D(int batch, int height, int width, int in_depth,
...
+                  .Attr("_input_shapes", attr_input_shape)
                   .Finalize(graph, &conv2d));
 
+  int conv2d_node_id = conv2d->id();
+  if (IsMKLEnabled()) {
+    std::unique_ptr<Graph>* ug = new std::unique_ptr<Graph>(graph);
+    RunMklLayoutRewritePass(ug);
+    // After we ran the pass conv2d node might be overwritten
+    // so we need to make sure that it is pointing to the one
+    // that exists
```
Comment (cantonios) [inline tensorflow/core/kernels/conv_ops_benchmark_test.cc:131]: nit: punctuation.

**Review by cantonios — APPROVED** [review]

File: tensorflow/core/kernels/conv_ops_benchmark_test.cc (hunk 1 above)
Comment (milpuz01) [reply tensorflow/core/kernels/conv_ops_benchmark_test.cc]: Thanks. Fixed in [a9de346](https://github.com/tensorflow/tensorflow/pull/60160/commits/a9de346c45eb2d0f8742d4b7997dd47ad76f043f)

**Review by cantonios — APPROVED** [review]

File: tensorflow/core/kernels/conv_ops_benchmark_test.cc [hunk 2]
```diff
@@ -97,19 +95,45 @@ static Conv2DGraph Conv2D(int batch, int height, int width, int in_depth,
...
                   .Attr("strides", {1, 1, 1, 1})
                   .Attr("padding", "SAME")
                   .Attr("data_format", ToString(data_format))
+                  .Attr("_input_shapes", attr_input_shape)
                   .Finalize(graph, &conv2d));
 
+  int conv2d_node_id = conv2d->id();
+  if (IsMKLEnabled()) {
+    std::unique_ptr<Graph>* ug = new std::unique_ptr<Graph>(graph);
+    RunMklLayoutRewritePass(ug);
```
Comment (cantonios) [inline tensorflow/core/kernels/conv_ops_benchmark_test.cc:128]: This needs to be guarded by `INTEL_MKL`, otherwise we see build failures.

**Review by cantonios — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/conv_ops_benchmark_test.cc (hunk 2 above)
Comment (milpuz01) [reply tensorflow/core/kernels/conv_ops_benchmark_test.cc:129]: Thanks. Fixed in 4433d4b

**Review by cantonios — APPROVED** [review]

**Pull request merged**